.github/workflows/airflow-plugin.yml

```diff
@@ -0,0 +1,84 @@
+name: airflow plugin build & test
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - ".github/workflows/airflow-plugin.yml"
+      - "metadata-ingestion-modules/airflow-plugin/**"
+      - "metadata-ingestion/**"
+      - "metadata-models/**"
+  pull_request:
+    branches:
+      - master
+    paths:
+      - ".github/**"
+      - "metadata-ingestion-modules/airflow-plugin/**"
+      - "metadata-ingestion/**"
+      - "metadata-models/**"
+  release:
+    types: [published]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  airflow-plugin:
+    runs-on: ubuntu-latest
+    env:
+      SPARK_VERSION: 3.0.3
+      DATAHUB_TELEMETRY_ENABLED: false
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.10"]
+        command: ["testQuick"] # add "lint" here after datahub_provider code is moved to metadata-ingestion-modules/airflow-plugin.
+        extraPythonRequirement:
```

Comment:
I am worried that this is going to be a lot of CI jobs: 2 py versions x 2 commands x 5 airflow versions -> 20 jobs

maybe we do airflow 2.0 and 2.2 test with python 3.7, and 2.4+ test with python 3.10? also we could combine the lint + test steps

Reply:
one drawback when lint and test steps are combined is that the combined action will be fail-fast, i.e. if lint fails then tests will not run but not a blocker as ultimately we will need both to pass.